notifier/notifier.go

```diff
@@ -158,84 +159,62 @@ func (n *Notifier) findTask(stopper *utils.Stopper) (string, database.Notificati
 		}
 
 		// Lock the notification.
-		if hasLock, _ := database.Lock(node, lockDuration, n.lockIdentifier); hasLock {
+		if hasLock, _ := database.Lock(node, lockDuration, whoAmI); hasLock {
 			log.Infof("found and locked a notification: %s", notification.GetName())
 			return node, notification
 		}
 	}
 }
 
-func (n *Notifier) handleTask(node string, notification database.Notification) bool {
+func handleTask(notification database.Notification, st *utils.Stopper, maxAttempts int) (bool, bool) {
 	// Get notification content.
 	// TODO(Quentin-M): Split big notifications.
 	notificationContent, err := notification.GetContent()
 	if err != nil {
 		log.Warningf("could not get content of notification '%s': %s", notification.GetName(), err)
-		return false
+		return false, false
 	}
 
 	// Create notification.
-	payload := Notification{
+	payload := &Notification{
 		Name:    notification.GetName(),
 		Type:    notification.GetType(),
 		Content: notificationContent,
 	}
 
-	// Marshal notification.
-	jsonPayload, err := json.Marshal(payload)
-	if err != nil {
-		log.Errorf("could not marshal content of notification '%s': %s", notification.GetName(), err)
-		return false
-	}
-
 	// Send notification.
-	resp, err := n.client.Post(n.endpoint, "application/json", bytes.NewBuffer(jsonPayload))
-	defer resp.Body.Close()
-	if err != nil || (resp.StatusCode != 200 && resp.StatusCode != 201) {
-		log.Errorf("could not send notification '%s': (%d) %s", notification.GetName(), resp.StatusCode, err)
-		return false
+	// TODO(Quentin-M): Backoff / MaxRetries
+	for notifierName, notifier := range notifiers {
+		for attempts, backOff := 0, time.Duration(0); ; attempts, backOff = attempts+1, timeutil.ExpBackoff(backOff, maxBackOff) {
```

Comment:
Let's make this slightly more readable by making it an `for {}` and doing the assignment around the loop.